tracer/test/Datadog.Trace.Security.IntegrationTests/Rcm/AspNetCore5AsmRemoteRules.cs

```diff
@@ -24,24 +25,39 @@ public AspNetCore5AsmRemoteRules(AspNetCoreTestFixture fixture, ITestOutputHelpe
 
         [SkippableFact]
         [Trait("RunOnWindows", "True")]
-        public async Task TestNewRemoteRules()
+        public async Task TestRemoteRules()
         {
             var url = "/Health/?[$slice]=value";
             await TryStartApp();
             var agent = Fixture.Agent;
             var settings = VerifyHelper.GetSpanVerifierSettings();
 
+            // Test new rules
             var spans1 = await SendRequestsAsync(agent, url);
-            var productId = nameof(TestNewRemoteRules);
+            var productId = nameof(TestRemoteRules);
             await agent.SetupRcmAndWait(Output, new List<(object Config, string ProductName, string Id)> { (GetRules("2.22.222"), "ASM_DD", productId) });
             var spans2 = await SendRequestsAsync(agent, url);
             await agent.SetupRcmAndWait(Output, new List<(object Config, string ProductName, string Id)> { (GetRules("3.33.333"), "ASM_DD", productId) });
             var spans3 = await SendRequestsAsync(agent, url);
 
+            // Test deletion + switch back to default rules when no RC no rules + keep updated blocking action + reset blocking actions
+            ResetDefaultUserAgent();
+            var block405Action = new Payload() { Actions = [new Action { Id = "block", Type = "block_request", Parameters = new Parameter { StatusCode = 405, Type = "json" } }] };
+            var block405ActionProductId = "action";
+            await agent.SetupRcmAndWait(Output, new List<(object Config, string ProductName, string Id)> { (block405Action, "ASM", block405ActionProductId), (GetNonBlockingRules(), "ASM_DD", productId) });
```

Comment:
NIT here we are testing actions as well, I wonder if we don't want to stick to testing only remote rules. I.e having another rule name kicking in and then the original one again in the snapshots, like just with a different id: `ua0-600-56x-remote` ,and then when removed it falls back to original.

Reply:
I think it's good to have extra tests that cover the rules being removed, but the rule added use the agent `dd-test-scanner-log-block`, which also appears in the default rule set, so it would be better use a different value.

Reply:
Also, I found it difficult to figure out what the expected output in the verify files should be, so it might be nice to explain that better here.

Reply:
Changes in f5ef3fe5dd5bd3d8e5913da169007ec0c4ff8dfc:
- the new rule id to `new-test-non-blocking` to easily spot the new triggered rule from the snapshot.
- added comments to better know what is expected on each span.